javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -0,0 +1,109 @@
+package java.util.regex
+
+import scalanative.native._, stdlib._, stdio._
+import scalanative.runtime.struct
+
+@extern
+object Regex {
+  def regcomp(regex: Ptr[Int], str: CString, num: Int): Int = extern
+  def regexec(regex: Ptr[Int],
+              str: CString,
+              nmatch: Int,
+              matchposition: Ptr[RegMatch.regmatch_t],
+              num2: Int): Int = extern
```

Comment:
It's important to use `CInt` instead of just `Int` here to keep it explicit that this type might have different size on some platforms (even though we only support 64-bit platforms at the moment.)